cylc/flow/task_events_mgr.py

```diff
@@ -864,7 +864,7 @@ def _setup_job_logs_retrieval(self, itask, event):
             str(itask.point), itask.tdef.name, itask.submit_num)
         events = (self.EVENT_FAILED, self.EVENT_RETRY, self.EVENT_SUCCEEDED)
         if (event not in events or
-                itask.platform['name'] == 'localhost' or
+                get_host_from_platform(itask.platform) == 'localhost' or
```

Comment:
I think there is a `is_remote_platform` function which performs this test a bit more safely (e.g. `localhost` might not be the only host in the platform)

Reply:
Ah doh, thanks I had made a mental note to do this - must start writing things down!